File: frontend/src/platform/analytics-context.ts
export type ComparisonMode =
  | 'none'
  | 'previous_period'
  | 'previous_month'
  | 'previous_year'
  | 'custom'

export type AnalyticsFilters = {
  channels: string[]
  categories: string[]
  skus: string[]
  couriers: string[]
  warehouses: string[]
  payment_methods: string[]
  states: string[]
  zones: string[]
}

export type AnalyticsPeriod = {
  start_date: string
  end_date: string
}

export type ComparisonPeriod = {
  mode: ComparisonMode
  start_date?: string | null
  end_date?: string | null
}

export type AnalyticsContext = {
  workspace_id?: string | null
  brand_id?: string | null

  period: AnalyticsPeriod

  comparison: ComparisonPeriod

  filters: AnalyticsFilters
}

export const emptyFilters: AnalyticsFilters = {
  channels: [],
  categories: [],
  skus: [],
  couriers: [],
  warehouses: [],
  payment_methods: [],
  states: [],
  zones: [],
}

export const defaultAnalyticsContext: AnalyticsContext = {
  period: {
    start_date: '2025-11-01',
    end_date: '2025-11-30',
  },

  comparison: {
    mode: 'previous_month',
  },

  filters: {
    ...emptyFilters,
  },
}

File: backend/app/analytics_context.py
from typing import Literal, Optional

from pydantic import BaseModel, Field


# ============================================================
# DATE RANGE
# ============================================================


class DateRange(BaseModel):
    start_date: str
    end_date: str


# ============================================================
# COMPARISON
# ============================================================


class ComparisonPeriod(BaseModel):
    mode: Literal[
        "none",
        "previous_period",
        "previous_month",
        "previous_year",
        "custom",
    ] = "previous_period"

    start_date: Optional[str] = None
    end_date: Optional[str] = None


# ============================================================
# GLOBAL FILTERS
# ============================================================


class AnalyticsFilters(BaseModel):
    channels: list[str] = Field(
        default_factory=list
    )

    categories: list[str] = Field(
        default_factory=list
    )

    skus: list[str] = Field(
        default_factory=list
    )

    couriers: list[str] = Field(
        default_factory=list
    )

    warehouses: list[str] = Field(
        default_factory=list
    )

    payment_methods: list[str] = Field(
        default_factory=list
    )

    states: list[str] = Field(
        default_factory=list
    )

    zones: list[str] = Field(
        default_factory=list
    )


# ============================================================
# GLOBAL ANALYTICS CONTEXT
# ============================================================


class AnalyticsContext(BaseModel):
    """
    Shared ProfitLens analytics context.

    This becomes the common contract used by:
    - dashboards
    - charts
    - tables
    - AI Analyst
    - investigations
    - Scenario Lab
    - exports

    Existing month-based endpoints can continue operating
    during the migration.
    """

    workspace_id: Optional[str] = None
    brand_id: Optional[str] = None

    period: DateRange

    comparison: ComparisonPeriod = Field(
        default_factory=ComparisonPeriod
    )

    filters: AnalyticsFilters = Field(
        default_factory=AnalyticsFilters
    )
